# 1. 94

[原文链接](https://www.topgoer.cn/docs/gomianshiti/mian1)

## 1.1. 问题

下面这段代码输出的内容

```go
package main

 import (
     "fmt"
 )

 func main() {
     defer_call()
 }

func defer_call() {
    defer func() { fmt.Println("打印前") }()
    defer func() { fmt.Println("打印中") }()
    defer func() { fmt.Println("打印后") }()
    panic("触发异常")
}
```

## 1.2. 答案

```
打印后
打印中
打印前
panic: 触发异常
```

* defer 的执行顺序是**后进先出**。
* 当出现 panic 语句的时候，会先按照 defer 的后进先出的顺序执行，最后才会执行 panic
